syntax = "proto3";

option go_package = "shadowify/gen";

package transcriber;

service TranscriberService {
  rpc Transcribe (TranscribeRequest) returns (TranscribeResponse);
}

message TranscribeRequest {
  string youtube_id = 1;
}

message Segment {
  string text = 1;
  float start_time = 2;
  float end_time = 3;
}

message TranscribeResponse {
  repeated Segment segments = 1;
}

